source

```diff
@@ -107224,6 +107259,121 @@ document.querySelector("button").addEventListener("click", bound);
    <li><p>Return <var>script</var>.</p></li>
   </ol>
 
+  <p>To <dfn data-x="creating a WebAssembly module script">create a WebAssembly module script</dfn>,
+  given a <span>byte sequence</span> <var>bodyBytes</var>, an <span>environment settings
+  object</span> <var>settings</var>, a <span>URL</span> <var>baseURL</var>, and a <span>script fetch
+  options</span> <var>options</var>:</p>
+
+  <ol>
+   <li><p>If <span data-x="concept-environment-noscript">scripting is disabled</span> for
+   <var>settings</var>, then set <var>bodyBytes</var> to the byte sequence 0x00 0x61 0x73 0x6d 0x01
+   0x00 0x00 0x00.</p></li>
+
+   <li><p>Let <var>script</var> be a new <span>module script</span> that this algorithm will
+   subsequently initialize.</p></li>
+
+   <li><p>Set <var>script</var>'s <span data-x="concept-script-settings-object">settings
+   object</span> to <var>settings</var>.</p></li>
+
+   <li><p>Set <var>script</var>'s <span data-x="concept-script-base-url">base URL</span> to
+   <var>baseURL</var>.</p></li>
+
+   <li><p>Set <var>script</var>'s <span data-x="concept-script-script-fetch-options">fetch
+   options</span> to <var>options</var>.</p></li>
+
+   <li><p>Set <var>script</var>'s <span data-x="concept-script-parse-error">parse error</span> and
+   <span data-x="concept-script-error-to-rethrow">error to rethrow</span> to null.</p></li>
+
+   <li>
+    <p>Let <var>result</var> be <span data-x="js-ParseModule">parse a WebAssembly module</span>(<var>source</var>,
+    <var>settings</var>'s <span data-x="environment settings object's realm">realm</span>,
+    <var>script</var>).</p>
```

Comment:
```suggestion
    <p>Let <var>result</var> be the result of <span data-x="...">parsing a WebAssembly module</span> given <var>bodyBytes</var>,
    <var>settings</var>'s <span data-x="environment settings object's realm">realm</span>,
    <var>script</var>).</p>
```


Where `data-x` must somehow link to https://webassembly.github.io/esm-integration/js-api/index.html#parse-a-webassembly-module.

Also note that "parse a WebAssembly module" expects an ArrayBuffer and not a bytes sequence, but maybe we could make it accept either.

Reply:
A byte sequence is a WhatWG concept not a W3C one so perhaps it would be better to explicitly pass an array buffer. I've gone ahead with that refactoring for now.

Reply:
The first line in "parse a WebAssembly module" calls https://webidl.spec.whatwg.org/#dfn-get-buffer-source-copy from WHATWG to get a byte sequence anyway, so there would be no problem with just using it.

Reply:
Ahh interesting - do you think we could change the ESM integration to take a byte sequence directly then, refactoring the callers to do this step? I guess I'm not 100% sure if there are outward consequences to that further?

Reply:
Done.